tests/unit/test_download.py

```diff
@@ -527,13 +527,14 @@ def test_http_cache_is_not_enabled(self, tmpdir):
 
         assert not hasattr(session.adapters["http://"], "cache")
 
-    def test_insecure_host_cache_is_not_enabled(self, tmpdir):
+    def test_insecure_host_adapter(self, tmpdir):
         session = PipSession(
             cache=tmpdir.joinpath("test-cache"),
             insecure_hosts=["example.com"],
         )
 
         assert not hasattr(session.adapters["https://example.com/"], "cache")
+        assert "https://example.com:" in session.adapters
```

Comment:
While you're at it, would it be just as easy to check that `https://example.com/` is a key, too?